cmake/GolangSimple.cmake

```diff
@@ -27,21 +27,4 @@ function(BUILD_GO_PROGRAM NAME MAIN_SRC CGO_CFLAGS CGO_LIBRARY_FLAGS)
 
   add_custom_target(${NAME}_all ALL DEPENDS ${NAME})
   install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/${NAME} DESTINATION bin)
-endfunction(BUILD_GO_PROGRAM)
-
-# Run go test for a subdirectory of interest
-function(RUN_GO_TEST NAME MAIN_SRC CGO_CFLAGS CGO_LIBRARY_FLAGS)
-  message(STATUS "${NAME}")
-  message(STATUS "${MAIN_SRC}")
-  get_filename_component(MAIN_SRC_ABS ${MAIN_SRC} ABSOLUTE)
-  message(STATUS "${MAIN_SRC_ABS}")
-
-  # This requires an output so we generate a .stamp to indicate it ran
-  add_custom_command(
-                    TARGET ${NAME}
-                    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-                    COMMAND go test -c -run types_test
-                    DEPENDS ${MAIN_SRC_ABS}
-                    COMMENT "Testing Go library")
-
-endfunction(RUN_GO_TEST)
```

Comment:
Im not 100% sure what this does since I am not as familiar with cake, but why delete it after adding it in form the last commit?

Reply:
This was an idea for a function that would run the go test by way of building a binary first. I decided against this approach because it adds huge complexity to something that is relatively simple. To be more explicit, the two options were:

- require building a binary for every library_test.go from here out into the future that is then run in a shard test
- OR just call go test directly from the shard test

The second is simpler, more sane, and in line with what go would do. I would leave this approach for now and we can discuss with others in the higher level PR if they want something different, and then look back to something like this if possible. I am very happy (imho) skipping extra cmake that isn't necessary.